File: php/features.php
<?php
declare(strict_types=1);

// PricingDay SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PricingDayFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PricingDayBaseFeature();
            case "test":
                return new PricingDayTestFeature();
            default:
                return new PricingDayBaseFeature();
        }
    }
}

File: py/features.py
# PricingDay SDK feature factory

from feature.base_feature import PricingDayBaseFeature
from feature.test_feature import PricingDayTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PricingDayBaseFeature(),
        "test": lambda: PricingDayTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
